### PR TITLE
move images from gcr.io to quay.io

### PR DESCRIPTION
**Pull request by coffeepac**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature
> /kind flake

**What this PR does / why we need it**:
moves images from being hosted in `gcr.io` to `quay.io`.  The reason for this move is that `gcr.io` isn't free, `quay.io` is and I don't have a personal budget to host these images.

Once this is merged I'm going to turn off access to the existing images.  If people need older images I can rebuild specific versions when asked.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by cblecker** [comment]
/hold

@coffeepac Can we deprecate this code completely? Is it used somewhere? It's concerning that binary images in k/k are in repos that aren't in the project's control.

cc: @kubernetes/sig-cluster-lifecycle-pr-reviews

**Comment by monotek** [comment]
@cblecker 

Also the current gcr.io docker repo (gcr.io/fluentd-elasticsearch/fluentd) is not in the projects control.
The problem now is that gcr.io is charging to much (private) money so we decided to move to a free registry (https://quay.io/organization/fluentd_elasticsearch).

The reason for the move away from the the original repo (gcr.io/google-containers/fluentd-elasticsearch) has been, that only Google employees we're able to build the images manually and we sometimes had to wait for several weeks to get this done, when we liked to release a new version of the image. 

/test pull-kubernetes-e2e-gce-100-performance

File: cluster/addons/fluentd-elasticsearch/es-image/Makefile [hunk 1]
```diff
@@ -12,11 +12,14 @@
...
+PREFIX = quay.io/fluentd_elasticsearch
 IMAGE = elasticsearch
 TAG = v6.7.2
 
 build:
-	gcloud builds submit --tag ${PREFIX}/${IMAGE}:${TAG}
+	docker build  --tag ${PREFIX}/${IMAGE}:${TAG} .
+
+push:
+    docker push ${PREFIX}/${IMAGE}:${TAG}
```
Comment (monotek) [inline cluster/addons/fluentd-elasticsearch/es-image/Makefile]: please add push with latest tag too...

File: cluster/addons/fluentd-elasticsearch/fluentd-es-image/Makefile [hunk 2]
```diff
@@ -12,11 +12,14 @@
...
+PREFIX = quay.io/fluentd_elasticsearch
 IMAGE = fluentd
 TAG = v2.5.2
 
 build:
-	gcloud builds submit --tag $(PREFIX)/$(IMAGE):$(TAG)
+	docker build  --tag ${PREFIX}/${IMAGE}:${TAG} .
+
+push:
+	docker push ${PREFIX}/${IMAGE}:${TAG}
```
Comment (monotek) [inline cluster/addons/fluentd-elasticsearch/fluentd-es-image/Makefile]: please add push with latest tag too...

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by neolit123** [comment]
/hold
/assign @spiffxp @dims 
PTAL WRT code organization - GCP / cluster + k8s-infra related.

**Comment by neolit123** [comment]
> What code do you want to deprecate? The whole monitoring stuff? Imho this is used a lot. The image is also used in the fluentd-elasticsearch helm chart: https://hub.helm.sh/charts/kiwigrid/fluentd-elasticsearch

`/cluster` is deprecated and tools like juju-charms moved out already.
if `/cluster/addons/fluentd-elasticsearch` is widely used then this is a signal for this addon to move out of the directory.

**Comment by monotek** [comment]
> `/cluster` is deprecated and tools like juju-charms moved out already.
> if `/cluster/addons/fluentd-elasticsearch` is widely used then this is a signal for this addon to move out of the directory.

What would be the right directory now?

**Comment by neolit123** [comment]
> What would be the right directory now?

juju moved out of k/k.

**Comment by monotek** [comment]
@piosz what are your thougts?

**Comment by detiber** [comment]
Assuming there is a SIG to provide sponsorship, `github.com/kubernetes-sigs` is probably the best home.

**Comment by coffeepac** [comment]
a. this is owned by sig-instrumentation.  
b. I have opened this feature request https://github.com/kubernetes/kubernetes/issues/79432 to make the move

however, I request both @cblecker and @neolit123 release their holds on this PR as I am spending money I shouldn't be to continue to host these images.  I do not have stats for how large an impact it would be if I cut off access to these images but it wouldn't be non-zero.  please see: https://github.com/kubernetes/kubernetes/issues/78684

**Comment by cblecker** [comment]
@coffeepac I'm poking some folks to see what we can do here. I don't want you to keep having to pay for things, but I also don't want to thrash where this image is located.

I understand this is urgent for cost reasons, and I'll be circling back with an answer ASAP.

**Comment by coffeepac** [comment]
thanks @cblecker.  this is entirely my own fault for incurring these charges as well.  I'm bad at reading fine print apparently.

**Review by monotek — APPROVED** [review]

**Comment by coffeepac** [comment]
@cblecker any progress on this or can we do a bit of location thrash while we sort out a long term solution?

**Comment by spiffxp** [comment]
/hold cancel
The wg-k8s-infra way (which I highly encourage and will eventually get pushy that someone use because it would also allow you to do this for "free") would be to request a staging repo, and using that for now.  https://github.com/kubernetes/k8s.io/tree/master/k8s.gcr.io#creating-staging-repos

This would also result in location thrash though, as we don't currently promote from staging repos to `k8s.gcr.io`.  So, let's thrash away.

**Comment by cblecker** [comment]
@coffeepac Immediate term, yeah, we can just move the reference location.

The next step after this is I'd like to have a plan for moving this code out, I think. The manifests look stale, this is just the image generation piece, while other pieces like the helm chart for installation https://github.com/kiwigrid/helm-charts/blob/master/charts/fluentd-elasticsearch/values.yaml#L2 exist elsewhere.

There's also a couple other references to the old image location in our code base: https://cs.k8s.io/?q=gcr.io%2Ffluentd-elasticsearch

WDYT?

**Comment by coffeepac** [comment]
Happy to move the code into a sig-inst repo.  I tried doing that 6 months ago and it wasn't possible for some reason, I think it was tied to the build process for the add-on manager?  But k8s infra has come along since then so I'm happy to do it.  

I'll get a plan onto the feature ticket early next week for discussion. 

Thanks for the help friends, I know this is a poor solution and I appreciate you taking it to help out my old poor decision.

**Comment by monotek** [comment]
If it helps I'm willing to move fluentd-elasticsearch helm chart out of the Kiwigrid repo. Is there already an official k8s org helm registry available somewhere or would we need to create one?

**Comment by vguaglione** [comment]
Where is this image on quay.io? I've tried the following:

docker pull quay.io/fluentd-elasticsearch/fluentd:v2.5.1
docker pull quay.io/fluentd-elasticsearch/fluentd:latest
docker pull quay.io/fluentd-elasticsearch/fluentd

and all three return the following error:

Error response from daemon: unauthorized: access to the requested resource is not authorized

**Comment by monotek** [comment]
https://quay.io/repository/fluentd_elasticsearch/fluentd?tab=tags

**Pull request merged**